TBAnnotationClustering-Swift/ViewController.swift

```diff
@@ -43,7 +43,7 @@ class ViewController: UIViewController, MKMapViewDelegate {
         let toRemove = NSMutableSet(set: before)
         toRemove.minusSet(after as Set<NSObject>)
         
-        dispatch_async(dispatch_get_main_queue()) { () -> Void in
+        NSOperationQueue().addOperationWithBlock() {
```

Comment:
You are creating a new NSOperationQueue and modifying the mapView while running on that new queue. The mapView, like all UIViews, should only be modified on the main queue/thread.